pkg/utils/cel/quantity_test.go

```diff
@@ -0,0 +1,284 @@
+	tests := []struct {
+		name       string
+		expression string
+		expected   bool
+	}{
+		{
+			name:       "compare tebibyte",
+			expression: `Quantity("1Ti") == Quantity("1Ti")`,
+			expected:   true,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1Ti") == Quantity("1024Gi")`,
+			expected:   true,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1") != Quantity("1000m")`,
+			expected:   false,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1Ti") > Quantity("1024Gi")`,
+			expected:   false,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1Ti") < Quantity("1024Gi")`,
+			expected:   false,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1Ti") >= Quantity("1024Gi")`,
+			expected:   true,
+		},
+		{
+			name:       "compare tebibyte to gibibyte",
+			expression: `Quantity("1Ti") <= Quantity("1024Gi")`,
+			expected:   true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			evaluateAndCompareToBool(t, tt.expression, tt.expected)
+		})
+	}
+}
+
+func evaluateAndCompareToFloat64(t *testing.T, expression string, expected float64) {
+	refVal, err := runCELExpression(t, expression)
+	if err != nil {
+		t.Errorf("failed to run CEL expression: %v", err)
+		return
+	}
+
+	actual, err := AsFloat64(refVal)
+	if err != nil {
+		t.Errorf("failed to convert to float64: %v", err)
+	}
+
+	const epsilon = 1e-9
+	if math.Abs(actual-expected) > epsilon {
+		t.Errorf("expected %v, got %v", expected, actual)
+	}
+}
+
+func evaluateAndCompareToBool(t *testing.T, expression string, expected bool) {
+	refVal, err := runCELExpression(t, expression)
+	if err != nil {
+		t.Errorf("failed to run CEL expression: %v", err)
+		return
+	}
+
+	switch v := refVal.(type) {
+	case types.Bool:
+		actual := bool(v)
+		if actual != expected {
+			t.Errorf("expected %v, got %v", expected, actual)
+		}
+	default:
+		t.Errorf("expected bool, got %T", refVal)
+	}
+}
+
+func evaluateAndShouldError(t *testing.T, expression string) {
+	_, err := runCELExpression(t, expression)
+	if err == nil {
+		t.Errorf("expected error, but got none")
+	}
+}
+
+func runCELExpression(t *testing.T, expression string) (ref.Val, error) {
+	return runCELExpressionWithData(t, expression, map[string]any{})
+}
+
+func runCELExpressionWithData(t *testing.T, expression string, data map[string]any) (ref.Val, error) {
+	types := slices.Clone(DefaultTypes)
+	conversions := slices.Clone(DefaultConversions)
+	funcs := maps.Clone(DefaultFuncs)
+	methods := maps.Clone(FuncsToMethods(DefaultFuncs))
+
+	env, err := NewEnvironment(EnvironmentConfig{
+		Types:       types,
+		Conversions: conversions,
+		Methods:     methods,
+		Funcs:       funcs,
+		Vars: map[string]any{
+			"node":      corev1.Node{},
+			"pod":       corev1.Pod{},
+			"container": corev1.Container{},
+		},
+	})
```

Comment:
```suggestion
	funcs := maps.Clone(DefaultFuncs)
	funcs[nowName] = []any{func() time.Time {
		return time.Date(2006, 1, 2, 15, 4, 5, 0, time.UTC)
	}}
	env, err := NewEnvironment(EnvironmentConfig{
		Types:       DefaultTypes,
		Conversions: DefaultConversions,
		Methods:     FuncsToMethods(funcs),
		Funcs:       funcs,
		Vars:        data,
	})
```

Make Now return a fixed value, and explicitly compare its value in funcs_test.go.